pkg/reportingplugins/mercury/aggregate_functions.go

```diff
@@ -85,25 +85,47 @@ func GetConsensusCurrentBlock(paos []ParsedAttributedObservation, f int) (hash [
 	if len(validPaos) < f+1 {
 		return nil, 0, 0, fmt.Errorf("fewer than f+1 observations have a valid current block (got: %d/%d)", len(validPaos), len(paos))
 	}
-	// pick the most common blockhash with at least f+1 votes
-	hash, err = getConsensusCurrentBlockHash(validPaos, f+1)
-	if err != nil {
-		return hash, 0, 0, errors.Wrap(err, "couldn't get consensus current block")
+
+	type block struct {
+		hash string
+		num  int64
+		ts   uint64
 	}
 
-	// pick the most common block number with at least f+1 votes
-	num, err = getConsensusCurrentBlockNum(validPaos, string(hash), f+1)
-	if err != nil {
-		return hash, num, 0, errors.Wrap(err, "couldn't get consensus current block")
+	m := map[block]int{}
+	maxCnt := 0
+	for _, pao := range validPaos {
+		b := block{string(pao.CurrentBlockHash), pao.CurrentBlockNum, pao.CurrentBlockTimestamp}
+		m[b]++
+		if cnt := m[b]; cnt > maxCnt {
+			maxCnt = cnt
+		}
 	}
 
-	// pick the most common block timestamp with at least f+1 votes
-	ts, err = getConsensusCurrentBlockTimestamp(validPaos, string(hash), num, f+1)
-	if err != nil {
-		return hash, num, ts, errors.Wrap(err, "couldn't get consensus current block")
+	if maxCnt < f+1 {
+		return nil, 0, 0, errors.New("no unique block with at least f+1 votes")
+	}
+
+	var blocks []block
+	for b, cnt := range m {
+		if cnt == maxCnt {
+			blocks = append(blocks, b)
+		}
 	}
+	sort.Slice(blocks, func(i, j int) bool {
+		if blocks[i].num == blocks[j].num && blocks[i].ts == blocks[j].ts {
+			// tie-break on hash, all else being equal
+			return blocks[i].hash < blocks[j].hash
+		} else if blocks[i].num == blocks[j].num {
+			// if block number is equal and timestamps differ, take the latest timestamp
+			return blocks[i].ts > blocks[j].ts
+		} else {
+			// if block number is different, take the higher block number
+			return blocks[i].num > blocks[j].num
+		}
```

Comment:
This might read a little easier if the comparison is encapsulated in a method on `block`:
```suggestion
		return blocks[i],less(blocks[j])
```
```go
func (b block) less(b2 block) bool {
	if b.num == b2.num && b.ts == b2.ts {
		// tie-break on hash, all else being equal
		return b.hash < b2.hash
	} else ...
}
```